From: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>
Subject: [PATCH] writeback: fix non-integrity write-back
Date: Wed, 26 May 2010 16:08:40 +0300

This is a fix for commit 7c8a3554. Note, I only compile-tested
this.

Signed-off-by: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>
---
 fs/fs-writeback.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/fs-writeback.c b/fs/fs-writeback.c
index ea8592b..0242855 100644
--- a/fs/fs-writeback.c
+++ b/fs/fs-writeback.c
@@ -920,7 +920,7 @@ long wb_do_writeback(struct bdi_writeback *wb, int force_wait)
 		if (force_wait)
 			work->args.sync_mode = args.sync_mode = WB_SYNC_ALL;

-		post_clear = WB_SYNC_ALL || args.sb_pinned;
+		post_clear = args.sync_mode == WB_SYNC_ALL || args.sb_pinned;

 		/*
 		 * If this isn't a data integrity operation, just notify
-- 
1.6.6.1

